README.md

```diff
@@ -154,6 +155,64 @@ $ python -c 'from subprocess import run; print(run(["openssl","rand","16"],captu
 
+### How to use *Active Directory* (or any other LDAP) authentication
+WARP supports authentication agains an LDAP srever. It is, the simplest way to allow your LDAP directory users login on your WARP instalation. 
+
+Enable LDAP auth is as simple as set *WARP_AUTH_LDAP* env variable to *'true'*. When enabled WARP will check user login and password via LDAP bind action and the list of authorithed groups (see *WARP_LDAP_GROUP_MAP* env variable). If Bind acction succedd and user belongs to authorized groups login is allowed. 
+
+Authorization is managed by adding user, on first login, to WARP auth database and assign the WARP group defined by *WARP_LDAP_GROUP_MAP*. 
+
+It is possible to exclude some users from using LDAP auth by adding them to *WARP_LDAP_EXCLUDED_USERS* list. Users on this list will check credentials via warp auth database ignoring LDAP AUTH configuration.
+
+Supported configurations to contact the LDAP server are:
+- **LDAP protocol:** *LDAP* or *LDAPS*. It is not recommend to use LDAP plain connections, on production enviroments.
+- **Authentication mechanims:** *SIMPLE* or *NTLM v2*.
+
+ **Enviroment variables:**
+* WARP_AUTH_LDAP: 'true'
+* WARP_LDAP_EXCLUDED_USERS: array of users login that are excluded form LDAP auth (to check login of this users Warp use password stored on warp DB).
+* WARP_LDAP_GROUP_MAP: Array of mapping between LDAP group and default asigned group on WARP. Order is important as only one group is assigned. First match is used.
+* WARP_LDAP_USER_CLASS: LDAP user objectclass.
+* WARP_LDAP_USER_ID_ATTRIBUTE: LDAP directory attribute to be compare with login.
+* WARP_LDAP_USER_NAME_ATTRIBUTE: LDAP directory attribute used to obtain name for the automatically created user on warp DB. 
+* WARP_LDAP_USER_GROUPS_ATTRIBUTE: LDAP directory attribute contaning group list.
+* WARP_LDAP_SEARCH_BASE: Base domain name to locate user and groups
+* WARP_LDAP_AUTH_SERVER: yourServerName/Ip    
+* WARP_LDAP_AUTH_SERVER_PORT: LDAP/LDAPS server port 
+* WARP_LDAP_AUTH_USE_LDAPS: True form LDAPS connection not defined or False for LDAP plain connection
+* WARP_LDAP_AUTH_TLS_VERSION: *'1.2'* for TLS1.2 not defined for TLS1
+* WARP_LDAP_AUTH_CIPHER: Set to a valid cipher for LDAPS server not defined *ECDHE-RSA-AES256-SHA384* is used.
+* WARP_LDAP_AUTH_VALIDATE_CERT: *False* for non certificate validation not defined or True for Certificate validation.
+* WARP_LDAP_AUTH_TYPE: NTLM/SIMPLE 
+* WARP_LDAP_AUTH_NTLM_DOMAIN: NTLM domain name is the prefix used for the login name when NTLM AUTH is enabled *DomainName\loginname*
+* LDAP_MATCHING_RULE_IN_CHAIN: Set to 'true' to include nested groups on ldap group search. For Active Directory set to 'true'. On other LDAP implementation check if your ldap implementation supports [LDAP_MATCHING_RULE_IN_CHAIN](https://ldapwiki.com/wiki/LDAP_MATCHING_RULE_IN_CHAIN).  
```

Comment:
Prefix is a good change to keep consistence :). 
We use DA as well. Setting to true make tha users added to groups added to the groups indicated in WARP_LDAP_GROUP_MAP, that why i say needted groups are considered. If is set to false only users directly added to the groups in WARP_LDAP_GROUP_MAP are considered valid for login.  If behaviour is not as espected we can check.

Reply:
I think you refer to Nested Groups right? I agree with your point, but don't know why in our AD infrastructure (2019 functional level) this option renders authentication to a non-working condition... No user would succeed authentication at all, even if it belongs directly to a group!
I'll try to emulate again using **ldapsearch** to see what's wrong, perhaps something specific to my environment. Either way in our implementation we do not need this feature :-)

Reply:
Perhaps implementation of has changed in Active directory. I don´t use 2019 but a previous version. Ones you have make your reseacrh If the problem is that is not supported on 2019 we can explain in Readme.md  a note indicating so. :)

Reply:
No, please just ignore my comments! I've manually tried it using **ldapsearch** and everything went fine, unfortunately I am unable to try it again on WARP because I have little to no time available for doing this :-/ I think it's safe to just leave this recommendation as you've written.